### ISSUE TITLE
Horovod in GPU-CPU nodes

### ISSUE THREAD

**Issue by pattasiago** [horovod/horovod#1825]
Is it possible to distribute pytorch/tensorflow training in heterogeneous nodes, i.e, for 2 nodes, one using cpu and the other one using gpu.

**Comment by nvcastet** [comment]
@pattasiago Horovod uses synchronized distributed deep learning. After each step, the workers communicates their gradients, so you will always be waiting for the slowest worker, in this case the cpu. So, you will lose the performance benefit of having GPUs.

**Comment by pattasiago** [comment]
I asked because when I tried to run "pytorch_mnist" example in these heterogeneous nodes I ended up with "Mismatched BROADCAST CPU/GPU device selection: One rank specified device CPU, but another rank specified device GPU.". In this case, I have 2 conteiners built, the cpu one was built with dockerfile.cpu image from the repository and the gpu one with dockerfile.gpu

**Comment by nvcastet** [comment]
@pattasiago It probably won't work right out of the box because we never saw a use case for it for the reason I mentioned: adding the CPU node will give you worse performance than having single GPU node. Does that make sense?

**Comment by pattasiago** [comment]
@nvcastet totally
